py/selenium/webdriver/firefox/webdriver.py

```diff
@@ -47,35 +47,99 @@ class WebDriver(RemoteWebDriver):
+        with different options.  They are mapped on to a capabilities
+        dictionary that is passed on to the remote end.
+
+        As some of the options, such as `firefox_profile` and
+        `firefox_options.profile` are mutually exclusive, precedence is
+        given from how specific the setting is.  `capabilities` is the
+        least specific keyword argument, followed by `firefox_options`,
+        followed by `firefox_binary` and `firefox_profile`.
+
+        In practice this means that if `firefox_profile` and
+        `firefox_options.profile` are both set, the selected profile
+        instance will always come from the most specific variable.
+        In this case that would be `firefox_profile`.  This will result in
+        `firefox_options.profile` to be ignored because it is considered
+        a less specific setting than the top-level `firefox_profile`
+        keyword argument.  Similarily, if you had specified a
+        `capabilities["firefoxOptions"]["profile"]` Base64 string,
+        this would rank below `firefox_options.profile`.
+
+        :param firefox_profile: Instance of ``FirefoxProfile`` object.
+        :param firefox_binary: Full path to the Firefox binary to use.
+        :param timeout: Time to wait for Firefox to launch when using
+            the extension connection.
+        :param capabilities: Dictionary of desired capabilities.
+        :param proxy: The proxy settings to us when communicating with
+            Firefox via the extension connection.
+        :param executable_path: Full path to override which geckodriver
+            binary to use for Firefox 47.0.1 and greater, which
+            defaults to picking up the binary from the system path.
+        :param firefox_options: Instance of ``options.Options``.
+        :param log_path: Where to log information from the driver.
+
+        """
+        self.binary = None
+        self.profile = None
+
+        if capabilities is None:
+            capabilities = DesiredCapabilities.FIREFOX.copy()
+
+        if capabilities.get("binary"):
+            self.binary = capabilities["binary"]
+
+        # firefox_options overrides capabilities
+        if firefox_options is not None:
+            self.binary = firefox_options.binary or self.binary
+            self.profile = firefox_options.profile or self.profile
+
+        # firefox_binary and firefox_profile
+        # override firefox_options
+        if firefox_binary is not None:
+            self.binary = firefox_binary
+            self.options.binary = firefox_binary
+        if firefox_profile is not None:
+            self.profile = firefox_profile
+            self.options.profile = firefox_profile
+
+        # disable native events if globally disabled
+        if self.profile is not None:
+            self.profile.native_events_enabled = (
+                self.NATIVE_EVENTS_ALLOWED and self.profile.native_events_enabled)
```

Comment:
I think you’re right this might matter when you’re _not_ using geckodriver. It is an intentional change not to pass a profile to geckodriver as it is able to create one itself.